src/decode.rs

```diff
@@ -3959,7 +3969,8 @@ unsafe fn setup_tile(
     ts.last_qidx = AtomicU8::new(frame_hdr.quant.yac);
     ts.last_delta_lf.fill_with(Default::default);
 
-    ts_c.msac = MsacContext::new(data, frame_hdr.disable_cdf_update != 0, &c.dsp.msac);
+    // SAFETY: `data` comes from `f.tiles`, which will live at least as long as the created `MsacContext`.
+    ts_c.msac = unsafe { MsacContext::new(data, frame_hdr.disable_cdf_update != 0, &c.dsp.msac) };
```

Comment:
Not sure if we should make this one safe yet.  The lifetimes are not super clear and it might be possible to make this safe.

Reply:
I think marking `setup_tile` safe for now is fine, though if we can make `MsacContext::new` safe at a later point that'd be great.

Reply:
Hmm, it doesn't matter too much I guess.  It's fine as is then.